trie/stacktrie.go

```diff
@@ -90,15 +89,13 @@ func (st *StackTrie) MarshalBinary() (data []byte, err error) {
 		w = bufio.NewWriter(&b)
 	)
 	if err := gob.NewEncoder(w).Encode(struct {
-		Nodetype  uint8
-		Val       []byte
-		Key       []byte
-		KeyOffset uint8
+		Nodetype uint8
+		Val      []byte
+		Key      []byte
```

Comment:
I need to lookup why we implemented binary marshalling, but if we somewhere do encode this to disk, then this is a breaking change.

Reply:
So, we added it here: https://github.com/ethereum/go-ethereum/pull/22685

@gballet do you recall why we/I did that?

Reply:
Thanks so much to help reach these accomplishments on the horizon